app/api/attendees.py

```diff
@@ -170,6 +170,17 @@ def before_update_object(self, obj, data, kwargs):
 #         if not has_access('is_registrar', event_id=obj.event_id):
 #         raise ForbiddenException({'source': 'User'}, 'You are not authorized to access this.')
 
+        if 'ticket' in data:
+            user = safe_query(self, User, 'id', current_identity.id, 'user_id')
+            ticket = db.session.query(Ticket).filter_by(
+                id=int(data['ticket']), deleted_at=None
+                ).first()
+            if not (user.is_verified or ticket.price):
```

Comment:
Condition should be not user.is_verified and ticket.price == 0

It should be clear